src/realm/object-store/object_store.cpp

```diff
@@ -762,23 +762,28 @@ static void apply_pre_migration_changes(Group& group, std::vector<SchemaChange>
 }
 
 enum class DidRereadSchema { Yes, No };
+enum class HandleBackLinksAutomatically { Yes, No };
 
 static void apply_post_migration_changes(Group& group, std::vector<SchemaChange> const& changes,
-                                         Schema const& initial_schema, DidRereadSchema did_reread_schema)
+                                         Schema const& initial_schema, DidRereadSchema did_reread_schema,
+                                         HandleBackLinksAutomatically handle_backlinks_automatically)
 {
     using namespace schema_change;
     struct Applier {
-        Applier(Group& group, Schema const& initial_schema, DidRereadSchema did_reread_schema)
+        Applier(Group& group, Schema const& initial_schema, DidRereadSchema did_reread_schema,
+                HandleBackLinksAutomatically handle_backlinks_automatically)
             : group{group}
             , initial_schema(initial_schema)
             , table(group)
             , did_reread_schema(did_reread_schema == DidRereadSchema::Yes)
+            , handle_backlinks_automatically(handle_backlinks_automatically == HandleBackLinksAutomatically::Yes)
```

Comment:
let's keep the default as the previous behaviour so we don't make a breaking change

Reply:
As I designed this stuff, you need to set a parameter in RealmConfig, which is by default set to false.
```bool automatic_handle_backlicks_in_migrations = false;``` in fact all the previous tests are OK, and they are using the old logic.